pkg/plugins/runtime/gateway/gateway_route_generator_test.go

```diff
@@ -0,0 +1,222 @@
+package gateway_test
+
+import (
+	"path"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/ghodss/yaml"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/core/runtime"
+	"github.com/kumahq/kuma/pkg/test/matchers"
+	xds_server "github.com/kumahq/kuma/pkg/xds/server/v3"
+)
+
+var _ = Describe("Gateway Gateway Route", func() {
+	var rt runtime.Runtime
+	var dataplanes *DataplaneGenerator
+
+	Do := func() (cache.Snapshot, error) {
+		serverCtx := xds_server.NewXdsContext()
+		reconciler := xds_server.DefaultReconciler(rt, serverCtx)
+
+		// We expect there to be a Dataplane fixture named
+		// "default" in the current mesh.
+		ctx, proxy := MakeGeneratorContext(rt,
+			core_model.ResourceKey{Mesh: "default", Name: "default"})
+
+		Expect(proxy.Dataplane.Spec.IsBuiltinGateway()).To(BeTrue())
+
+		if err := reconciler.Reconcile(*ctx, proxy); err != nil {
+			return cache.Snapshot{}, err
+		}
+
+		return serverCtx.Cache().GetSnapshot(proxy.Id.String())
+	}
+
+	BeforeEach(func() {
+		var err error
+
+		rt, err = BuildRuntime()
+		Expect(err).To(Succeed(), "build runtime instance")
+
+		Expect(StoreNamedFixture(rt, "mesh-default.yaml")).To(Succeed())
+		Expect(StoreNamedFixture(rt, "dataplane-default.yaml")).To(Succeed())
+		Expect(StoreNamedFixture(rt, "gateway-default.yaml")).To(Succeed())
+
+		dataplanes = &DataplaneGenerator{Manager: rt.ResourceManager()}
+
+		// TODO(jpeach) dataplane resources won't be used in the
+		// tests until the GatewayRoute generator starts generating
+		// endpoints.
+		dataplanes.Generate("echo-service")
+	})
+
+	DescribeTable("generate matching resources",
+		func(goldenFileName string, fixtureResources ...string) {
+			// given
+			for _, resource := range fixtureResources {
+				Expect(StoreInlineFixture(rt, []byte(resource))).To(Succeed())
+			}
+
+			// when
+			snap, err := Do()
```

Comment:
Nit: I would have just put the content of Do inline but that gets down to the "long function vs short function" debate that's not worth having :)

Reply:
WDYT of the DescribeTable? I find tests easier to read from an outside POV but that might just be me.

Reply:
Yeh, DescribeTable was a good change. I find it less readable with a large inline test function, so prefer to keep `Do()` for now.